src/main/scala/ScalaModulePlugin.scala

```diff
@@ -3,56 +3,85 @@ import Keys._
+  val scalaVersionsByJvm  = settingKey[Map[Int, List[(String, Boolean)]]]("For a Java major version (6, 8, 9), a list of a Scala version and a flag indicating whether to use this combination for publishing.")
 
-  private val canRunMima         = taskKey[Boolean]("Decides if MiMa should run.")
-  private val runMimaIfEnabled   = taskKey[Unit]("Run MiMa if mimaPreviousVersion and the module can be resolved against the current scalaBinaryVersion.")
+  // Settings applied to the entire build when the plugin is loaded.
 
-  lazy val scalaModuleSettings: Seq[Setting[_]] = Seq(
-    repoName            := name.value,
-
-    mimaPreviousVersion := None,
+  override def trigger = allRequirements
 
-    scalaVersionsByJvm  := Map.empty,
+  override def buildSettings: Seq[Setting[_]] = Seq(
+    scalaVersionsByJvm := Map.empty,
 
-    crossScalaVersions  := {
+    crossScalaVersions in ThisBuild := {
       val OneDot = """1\.(\d).*""".r // 1.6, 1.8
       val Maj    = """(\d+).*""".r   // 9
       val javaVersion = System.getProperty("java.version") match {
         case OneDot(n) => n.toInt
         case Maj(n)    => n.toInt
         case v         => throw new RuntimeException(s"Unknown Java version: $v")
       }
-      val isTravisPublishing = Option(System.getenv("TRAVIS_TAG")).getOrElse("").trim.nonEmpty
-      val scalaVersions = scalaVersionsByJvm.value.getOrElse(javaVersion, Nil) collect {
+
+      val isTravis = Option(System.getenv("TRAVIS")).exists(_ == "true") // `contains` doesn't exist in Scala 2.10
+      val isTravisPublishing = Option(System.getenv("TRAVIS_TAG")).exists(_.trim.nonEmpty)
+
+      val byJvm = scalaVersionsByJvm.value
+      if (byJvm.isEmpty)
+        throw new RuntimeException(s"Make sure to define `scalaVersionsByJvm in ThisBuild` in `build.sbt` in the root project, using the `ThisBuild` scope.")
+
+      val scalaVersions = byJvm.getOrElse(javaVersion, Nil) collect {
         case (v, publish) if !isTravisPublishing || publish => v
       }
       if (scalaVersions.isEmpty) {
-        if (isTravisPublishing) {
+        if (isTravis) {
           sLog.value.warn(s"No Scala version in `scalaVersionsByJvm` in build.sbt needs to be released on Java major version $javaVersion.")
           // Exit successfully, don't fail the (travis) build. This happens for example if `openjdk7`
           // is part of the travis configuration for testing, but it's not used for releasing against
           // any Scala version.
           System.exit(0)
         } else
-          throw new RuntimeException(s"No Scala version for Java major version $javaVersion. Adjust `scalaVersionsByJvm` in build.sbt.")
+          throw new RuntimeException(s"No Scala version for Java major version $javaVersion. Change your Java version or adjust `scalaVersionsByJvm` in build.sbt.")
       }
       scalaVersions
     },
 
-    scalaVersion       := crossScalaVersions.value.head,
+    scalaVersion in ThisBuild := crossScalaVersions.value.head
```

Comment:
JFYI the `in ThisBuild` is redundant here by virtue of this being in `buildSettings`.

That's also why the RHS here isn't `(crossScalaVersions in ThisBuild).value.head` because `buildSettings` maps the RHS too.